lib/utils.pm

```diff
@@ -69,10 +69,11 @@ sub wait_boot {
     }
     # reconnect s390
     elsif (check_var('ARCH', 's390x')) {
+        my $login_ready = qr/Welcome to SUSE Linux Enterprise Server.*\(s390x\)/;
```

Comment:
you can't use this in a variable, I had the same idea, but if you look closely, the Welcome message differs on zVM and zKVM...

If you find a workaround for that, feel free ;)